pandas/_libs/missing.pyx

```diff
@@ -290,12 +291,22 @@ def _create_binary_propagating_op(name, divmod=False):
 
     def method(self, other):
         if (other is C_NA or isinstance(other, str)
-                or isinstance(other, (numbers.Number, np.bool_))):
+                or isinstance(other, (numbers.Number, np.bool_, np.int64, np.int_))
+                or isinstance(other, np.ndarray) and not other.shape):
```

Comment:
This seems a bit strange, but I suppose it is to follow numpy behaviour of 0-dim arrays returning scalars from comparison operations? (if so, maybe add a comment about that)

Reply:
> to follow numpy behaviour of 0-dim arrays returning scalars from comparison operations?

It's to NumPy scalars. Without this, we'd have

```python
np.int64(1) == pd.NA
```

raise with

```pytb
>   out[:] = NA
E   IndexError: too many indices for array
```

Reply:
A numpy scalar (which is different from a 0-dim array) is not a ndarray, so wouldn't pass the `isinstance(other, np.ndarray)` test. So I don't fully understand how that example is related.

Reply:
could call lib.item_from_zerodim at the top?

Reply:
With this

```diff
diff --git a/pandas/_libs/missing.pyx b/pandas/_libs/missing.pyx
index 8d4d2c5568..e2bb6448cd 100644
--- a/pandas/_libs/missing.pyx
+++ b/pandas/_libs/missing.pyx
@@ -295,8 +295,7 @@ def _create_binary_propagating_op(name, is_divmod=False):
 
     def method(self, other):
         if (other is C_NA or isinstance(other, str)
-                or isinstance(other, (numbers.Number, np.bool_))
-                or isinstance(other, np.ndarray) and not other.shape):
+                or isinstance(other, (numbers.Number, np.bool_))):
             # Need the other.shape clause to handle NumPy scalars,
             # since we do a setitem on `out` below, which
             # won't work for NumPy scalars.
```

I have

```pytb
In [3]: np.int64(1) == pd.NA
/Users/taugspurger/.virtualenvs/pandas-dev/bin/ipython:1: DeprecationWarning: elementwise comparison failed; this will raise an error in the future.
  #!/Users/taugspurger/Envs/pandas-dev/bin/python
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-3-209227154583> in <module>
----> 1 np.int64(1) == pd.NA

~/sandbox/pandas/pandas/_libs/missing.pyx in pandas._libs.missing._create_binary_propagating_op.method()
    307         elif isinstance(other, np.ndarray):
    308             out = np.empty(other.shape, dtype=object)
--> 309             out[:] = NA
    310
    311             if is_divmod:

IndexError: too many indices for array
```

So when we get there, it really does seem like `np.int64(1)` is an ndarray. Is cython doing something?

Reply:
> So when we get there, it really does seem like np.int64(1) is an ndarray. Is cython doing something?

Possibly, no idea. But I trust you this is indeed needed to get it working